src/AbstractAlgebra.jl

```diff
@@ -409,7 +409,7 @@ macro show_name(io, O)
         set_name!(o, s)
       end
     end
-    if get(i, :compact, false) && s !== nothing
+    if get(i, :supercompact, false) && get(i, :compact, false) && s !== nothing
```

Comment:
```suggestion
    if get(i, :supercompact, false) || get(i, :compact, false) && s !== nothing
```
I am not sure if this is doing what we expect. If I understood https://github.com/Nemocas/AbstractAlgebra.jl/issues/1564 correctly, one of `:supercompact` and `:compact` should be `true`. This would imply the use of `||` here (see suggestion above).

Reply:
You are right. I don't know what I was thinking.